File: Lab3/3.1/Clock/Clock.cs
namespace Clock
{
    public class Clock
    {
        Counter _second = new("seconds");
        Counter _minute = new("minutes");
        Counter _hour = new("hours");

        public void Tick()
        {
            _second.Increment();
            if (_second.Tick > 59)
            {
                _second.Reset();
                _minute.Increment();

                if (_minute.Tick > 59)
                {
                    _minute.Reset();
                    _hour.Increment();

                    if (_hour.Tick > 23)
                    {
                        _hour.Reset();
                    }
                }
            }
        }

        public void ResetClock()
        {
            _second.Reset();
            _minute.Reset();
            _hour.Reset();
        }

        public string Time()
        {

            return _hour.Tick.ToString("00") + ":" + _minute.Tick.ToString("00") + ":" + _second.Tick.ToString("00");

        }
    }
}

File: Lab11/clock.py
class Clock:
    def __init__(self, hours=0, minutes=0, seconds=0):
        self.hours = hours
        self.minutes = minutes
        self.seconds = seconds

    def tick(self):
        self.seconds += 1
        if self.seconds >= 60:
            self.seconds = 0
            self.minutes += 1
            if self.minutes >= 60:
                self.minutes = 0
                self.hours += 1
                if self.hours >= 24:
                    self.hours = 0

    def __str__(self):
        return "{:02d}:{:02d}:{:02d}".format(self.hours, self.minutes, self.seconds)